File: src/main/java/rars/venus/settings/editor/BottomRowComponent.java
package rars.venus.settings.editor;

import org.jetbrains.annotations.NotNull;

import javax.swing.*;

public final class BottomRowComponent extends JPanel {
    public @NotNull JButton applyButton, applyAndCloseButton, cancelButton;

    public BottomRowComponent() {
        super();
        this.setLayout(new BoxLayout(this, BoxLayout.X_AXIS));
        this.applyButton = new JButton("Apply");
        this.applyAndCloseButton = new JButton("Apply and close");
        this.cancelButton = new JButton("Cancel");

        this.add(Box.createHorizontalGlue());
        this.add(this.cancelButton);
        this.add(Box.createHorizontalStrut(10));
        this.add(this.applyButton);
        this.add(this.applyAndCloseButton);
    }
}

File: src/main/java/rars/venus/settings/editor/BottomRowComponent.kt
package rars.venus.settings.editor

import rars.venus.util.BoxLayout
import javax.swing.BoxLayout
import javax.swing.JButton
import javax.swing.JPanel

class BottomRowComponent : JPanel() {
    val applyButton = JButton("Apply")
    val applyAndCloseButton = JButton("Apply and close")
    val cancelButton = JButton("Cancel")

    init {
        BoxLayout(BoxLayout.X_AXIS) {
            horizontalGlue()
            +cancelButton
            horizontalStrut(10)
            +applyButton
            +applyAndCloseButton
        }
    }
}
